src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Account/Pages/Manage/ChangePassword.razor

```diff
@@ -53,6 +53,11 @@
     protected override async Task OnInitializedAsync()
     {
         user = await UserAccessor.GetRequiredUserAsync(HttpContext);
+        if (user is null)
+        {
+            return;
+        }
```

Comment:
The `GetRequiredUserAsync()` naming suggests that an exception will be thrown if the user does not exist. This no longer happens, since `NavigateTo()` no longer throws by default, and it might not be clear that `GetRequiredUserAsync()` performs a redirect as a possible side effect.

Maybe we could rename that method to `GetUserOrRedirectAsync()`? Another idea is to remove `UserAccessor` entirely and add some kind of `RedirectToInvalidUser()` helper in `RedirectManager`:

```csharp
user = await UserManager.GetUserAsync(context.User);
if (user is null)
{
    RedirectManager.RedirectToInvalidUser();
    return;
}
```

Reply:
I like the approach with `RedirectManager.RedirectToInvalidUser`. In fact, it will just involve chopping up `GetRequiredUserAsync` in two parts because it's already redirecting to "Account/InvalidUser" in case of failure.

https://github.com/dotnet/aspnetcore/blob/0230498dfccaef6f782a5e37c60ea505081b72bf/src/ProjectTemplates/Web.ProjectTemplates/content/BlazorWeb-CSharp/BlazorWeb-CSharp/Components/Account/IdentityUserAccessor.cs#L14

Having that, all `throw new InvalidOperationException("User is not loaded")` can be replaced with `RedirectManager.RedirectToInvalidUser();`.